src/main/java/org/datadog/jmxfetch/util/CustomLogger.java

```diff
@@ -1,123 +1,166 @@
+        };
+
+        // prepare the different handlers
+        // --
+
+        ConsoleHandler stdoutHandler = null;
+        ConsoleHandler stderrHandler = null;
+        FileHandler fileHandler = null;
+
+        // the logLocation isn't always containing a file, it is sometimes
+        // referring to a standard output. We want to create a FileHandler only
+        // if the logLocation is a file on the FS.
+        if (logLocation != null && logLocation.length() > 0) {
+            if (!isStdOut(logLocation) && !isStdErr(logLocation)) {
+                // file logging
+                try {
+                    // maximum one 5MB file
+                    fileHandler = new FileHandler(logLocation, 5 * 1024 * 1024, 1);
+                    fileHandler.setFormatter(formatter);
+                } catch (Exception e) {
+                    fileHandler = null;
+                    log.error("can't open the file handler");
+                }
+            } else if (isStdErr(logLocation)) {
+                // console handler sending on stderr
+                // note that ConsoleHandler is sending on System.err
+                stderrHandler = new ConsoleHandler();
+                stderrHandler.setFormatter(formatter);
+                stderrHandler.setLevel(level.toJulLevel());
             }
         }
 
-        // replace default appender with the correct layout
-        LoggerConfig loggerConfig = config.getLoggerConfig(LogManager.ROOT_LOGGER_NAME);
-        loggerConfig.removeAppender(target);
+        // always have a console handler sending the logs to stdout
+        stdoutHandler = new StdoutConsoleHandler();
+        stdoutHandler.setFormatter(formatter);
+        stdoutHandler.setLevel(level.toJulLevel());
 
-        Appender appender = ConsoleAppender.newBuilder()
-                    .setConfiguration(config)
-                    .withName(target)
-                    .withLayout(layout)
-                    .build();
-        appender.start();
+        // should configure the root logger
+        // ---
 
-        loggerConfig.addAppender(appender, null, null);
-        loggerConfig.setLevel(level);
+        Logger logger = Logger.getLogger("");
 
-        ctx.updateLoggers();
+        // clean all existing handlers
+        for (Handler handler : logger.getHandlers()) {
+            logger.removeHandler(handler);
+        }
+
+        // set our configured handlers
+
+        if (fileHandler != null) {
+            logger.addHandler(fileHandler);
+        }
+        if (stdoutHandler != null) { // always non-null but doesn't cost much
+            logger.addHandler(stdoutHandler);
+        }
+        if (stderrHandler != null) {
+            logger.addHandler(stderrHandler);
+        }
+    }
+
+    /** closeHandlers closes all opened handlers. */
+    public static synchronized void shutdown() {
+        Logger logger = Logger.getLogger("");
+        for (Handler handler : logger.getHandlers()) {
+            handler.close();
```

Comment:
We likely want to remove the handlers first from the logger. We should also flush the handlers before closing I think?

Reply:
> The close method will perform a flush and then close the Handler

I'll remove the handler though 👍